tests/WeibullDistributionTest.php

```diff
@@ -0,0 +1,172 @@
+<?php
+require_once dirname(__FILE__) . "/../src/gburtini/Distributions/Weibull.php";
+
+use gburtini\Distributions\Weibull;
+
+class WeibullDistributionTest extends PHPUnit_Framework_TestCase
+{
+	public function test_Weibull_InstantiateDistribution() {
+		$distribution1 = new Weibull(2.0, 3.1);
+	}
+
+	public function test_Weibull_InvalidInstantiation() {
+		$this->setExpectedException('InvalidArgumentException');
+		$invalid = new Weibull(0, 1);
+	}
+
+	public function test_Weibull_InvalidInstantiation2() {
+		$this->setExpectedException('InvalidArgumentException');
+		$invalid = new Weibull(1, 0);
+	}
+
+	public function test_Weibull_InvalidInstantiation3() {
+		$this->setExpectedException('InvalidArgumentException');
+		$invalid = new Weibull(1, -1);
+	}
+
+	public function test_Weibull_InvalidInstantiation4() {
+		$this->setExpectedException('InvalidArgumentException');
+		$invalid = new Weibull(-1, 1);
+	}
+
+	public function test_Weibull_InvalidInstantiation5() {
+		$this->setExpectedException('InvalidArgumentException');
+		$invalid = new Weibull("a", 2);
+	}
+
+	public function testObjectDraw() {
+		mt_srand(1);
+
+		$d = new Weibull(3.2, 1.4);
+
+		$scale = 50000;
+		$cutoff = 0.9;
+		$counter = 0;
+
+		$draws = new SplFixedArray($scale);
+		for($i = 0; $i < $scale; $i++) {
+			$x = $d->rand();
+			$draws[$i] = $x;
+
+			if ($x > $cutoff) $counter = $counter + 1;
+		}
+
+		$number = array_sum((array) $draws) / count($draws);
+		$mean = $d->mean();
+		$this->assertEquals($number, $mean, "Attempting to draw from Weibull(3.2, 1.4)) {$scale} times gives us a value too far from the expected mean. This could be just random chance.", 0.001);
+
+		$p = $counter / $scale;
+		$this->assertEquals($p, 1 - $d->cdf($cutoff), "Attempting to draw from  Weibull(3.2, 1.4))  {$scale} times gives the wrong number of values greater than {$cutoff}. This could be just random chance.", 0.001);
+	}
+
+
+
+	public function test_Weibull_PDF() {
+		$accuracy = 1e-10;
```

Comment:
I guess we should strive for close to machine epsilon accuracy. I'll have a look at all the tests for this.